src/routers/course.ts

```diff
@@ -1,75 +1,71 @@
 import express from 'express';
-import Course from '../models/course';
-import { create, read, update, del } from '../operators/CourseOperations';
+import Ajv from 'ajv';
+import schema from '../schema.json';
 
 const courseRouter = express.Router();
+const ajv = new Ajv({ removeAdditional: true });
+
+const schemas = schema.components.schemas;
+const checkPost = ajv.compile(schemas.CourseCreate);
+const checkPut = ajv.compile(schemas.CourseUpdate);
+
+const ERROR_RESPONSE = 'Course not found.';
 
 courseRouter.get('/:id', async (req, res) => {
 	try {
 		const id = req.params.id;
 		const userID = req.header('userID');
 
-		if (id && userID) {
-			const course = await read(id, userID);
-			if (course) {
-				const resData = {
-					...course,
-					courseItems: Array.from(course.courseItems || []),
-				};
-				res.send(resData);
-			} else {
-				throw 'ERROR - course undefined';
-			}
-		} else {
-			throw 'ERROR - id undefined';
-		}
+		if (!id || !userID) throw Error(`invalid ${id ? 'user' : 'course'} id`);
+
+		// TODO: call db operation
```

Comment:
Add a note for the create methods that we will want to make sure that we set the `body.ownerId` property to `userID` as we don't want people to be able to create items for other people.